src/libraries/Common/tests/System/Net/Sockets/SocketTestExtensions.cs

```diff
@@ -49,5 +51,32 @@ public static (Socket, Socket) CreateConnectedSocketPair()
 
             return (client, server);
         }
+
+        // Tries to connect within the provided timeout interval
+        // Useful to speed up "can not connect" assertions on Windows
+        public static bool TryConnect(this Socket socket, EndPoint remoteEndpoint, int millisecondsTimeout)
+        {
+            using var mre = new ManualResetEvent(false);
```

Comment:
Nit: doesn't matter as much for tests, but it'd be cheaper to use ManualResetEventSlim

Reply:
Probably better not to dispose of this. You currently have a race condition where you could dispose of it and then set it.

Reply:
Re MRE:
Initially I wanted to use the slim variant, but I was afraid that the spinning could make things worse. Isn't `300ms` too long for `ManualResetEventSlim`? Any resources on guidance about how to choose?

Re Dispose: 
See my next comment.

Reply:
> I was afraid that the spinning could make things worse. Isn't 300ms too long for ManualResetEventSlim?

MRES doesn't just spin; after a tiny amount of spinning it'll fall back to an actual kernel wait.

If you're actually concerned about it (and I don't think you need to be), you can pass `spinCount: 0` to the ctor.